Content.Shared/Item/ItemToggle/SharedItemToggleSystem.cs

```diff
@@ -0,0 +1,313 @@
+        {
+            UpdateItemComponent(uid, itemToggleSize);
+        }
+        if (TryComp(uid, out ItemToggleMeleeWeaponComponent? itemToggleMelee))
+        {
+            UpdateWeaponComponent(uid, itemToggleMelee);
+        }
+        UpdateAppearance(uid, itemToggle);
+        UpdateLight(uid, itemToggle);
+
+        Dirty(uid, itemToggle);
+    }
+
+    /// <summary>
+    /// Used for items that require to be wielded in both hands to activate. For instance the dual energy sword will turn off if not wielded.
+    /// </summary>
+    private void TurnOffonUnwielded(EntityUid uid, ItemToggleComponent itemToggle, ItemUnwieldedEvent args)
+    {
+        if (itemToggle.Activated)
+            TryDeactivate(uid, args.User, itemToggle: itemToggle);
+    }
+
+    /// <summary>
+    /// Wieldable items will automatically turn on when wielded.
+    /// </summary>
+    private void TurnOnonWielded(EntityUid uid, ItemToggleComponent itemToggle, ref ItemWieldedEvent args)
+    {
+        if (!itemToggle.Activated)
+            TryActivate(uid, itemToggle: itemToggle);
+    }
+
+    /// <summary>
+    /// Used to update item appearance.
+    /// </summary>
+    private void UpdateAppearance(EntityUid uid, ItemToggleComponent itemToggle)
+    {
+        if (!TryComp(uid, out AppearanceComponent? appearance))
+            return;
+
+        _appearance.SetData(uid, ToggleableLightVisuals.Enabled, itemToggle.Activated, appearance);
+        _appearance.SetData(uid, ToggleVisuals.Toggled, itemToggle.Activated, appearance);
+    }
+
+    /// <summary>
+    /// Used to update light settings.
+    /// </summary>
+    private void UpdateLight(EntityUid uid, ItemToggleComponent itemToggle)
+    {
+        if (!_light.TryGetLight(uid, out var light))
+            return;
+
+        _light.SetEnabled(uid, itemToggle.Activated, light);
+    }
+
+    /// <summary>
+    /// Used to update weapon component aspects, like hit sounds, damage values and hidden status while activated.
+    /// </summary>
+    private void UpdateWeaponComponent(EntityUid uid, ItemToggleMeleeWeaponComponent itemToggleMelee)
+    {
+        if (!TryComp(uid, out MeleeWeaponComponent? meleeWeapon))
+            return;
+
+        //Sets the damage values to the item's default if none is stated.
+        itemToggleMelee.ActivatedDamage ??= meleeWeapon.Damage;
+        itemToggleMelee.DeactivatedDamage ??= meleeWeapon.Damage;
+        //Sets the no damage sound to the item's default if none is stated.
+        itemToggleMelee.ActivatedSoundOnHitNoDamage ??= meleeWeapon.NoDamageSound;
+        itemToggleMelee.DeactivatedSoundOnHitNoDamage ??= meleeWeapon.NoDamageSound;
+        //Sets the swing sound to the item's default if none is stated.
+        itemToggleMelee.ActivatedSoundOnSwing ??= meleeWeapon.SwingSound;
+        itemToggleMelee.DeactivatedSoundOnSwing ??= meleeWeapon.SwingSound;
+
+
+        if (IsActivated(uid))
+        {
```

Comment:
You should add a resolve component for activatable in UpdateWeaponComponent so you can pass the component in here as getting components isn't free.

Reply:
I don't think this is necessary anymore. Moved UpdateWeaponComponent to the MeleeWeapon system (now named OnItemToggle) and it runs on raised event, which has the activation state as an argument. As such, you don't get the ItemToggle component called at all anymore.